### PR TITLE
[stdlib] Move _stdlib_AtomicInt and friends out of the stdlib

### PR DESCRIPTION
**Pull request by moiseev**
Part 2 of <rdar://problem/45746339>

Contains changes from https://github.com/apple/swift/pull/20223

### CODE REVIEW

**Comment by moiseev** [comment]
@swift-ci Please test

**Review by milseman — APPROVED** [review]
Awesome, LGTM

**Comment by moiseev** [comment]
@swift-ci Please test

**Comment by moiseev** [comment]
@swift-ci Please test

File: stdlib/private/SwiftPrivate/CMakeLists.txt [hunk 1]
```diff
@@ -1,7 +1,13 @@
+set(swift_swiftprivate_compile_flags
+    "-parse-stdlib"
+    "-Xfrontend" "-disable-access-control")
```
Comment (jrose-apple) [inline stdlib/private/SwiftPrivate/CMakeLists.txt]: Why `-disable-access-control`?

File: stdlib/private/SwiftPrivate/AtomicInt.swift.gyb [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+#if arch(i386) || arch(arm)
+  let value = _swift_stdlib_atomicFetch${operation}Int32(
+    object: rawTarget.assumingMemoryBound(to: Int32.self),
+    operand: Int32(operand))
+#elseif arch(x86_64) || arch(arm64) || arch(powerpc64) || arch(powerpc64le) || arch(s390x)
+  let value = _swift_stdlib_atomicFetch${operation}Int64(
+    object: rawTarget.assumingMemoryBound(to: Int64.self),
+    operand: Int64(operand))
+#endif
+  return Int(value)
```
Comment (moiseev) [inline stdlib/private/SwiftPrivate/AtomicInt.swift.gyb]: @jrose-apple This is the reason for `-disable-access-control`. It's an internal initializer from a `Builtin` type.

**Pull request merged**